pyintegration/deephaven2/parquet.py

```diff
@@ -0,0 +1,206 @@
+        _JParquetTools.deleteTable(_get_file_object(path))
+    except Exception as e:
+        raise DHError(e, "failed to delate a parquet table on disk.") from e
+
+
+def write_table(table: Table, destination: str, col_definitions: List[Column] = None,
+                col_instructions: List[ColumnInstruction] = None, compression_codec_name: str = None,
+                max_dictionary_keys: int = None):
+    """ Write a table to a Parquet file.
+
+    Args:
+        table (Table): the source table
+        destination (str): destination file path; the file name should end in ".parquet" extension. If the path includes
+              non-existing directories they are created. If there is an error any intermediate directories previously
+              created are removed; note this makes this method unsafe for concurrent use
+        col_definitions (List[Column]): the column definitions to use
+        col_instructions (List[ColumnInstruction]): instructions for customizations while writing
+        compression_codec_name (str): the default compression codec to use, default is SNAPPY
+        max_dictionary_keys (int): the maximum dictionary keys allowed
+
+    Raises:
+        DHError
+    """
+    write_instructions = _build_parquet_instructions(col_instructions=col_instructions,
+                                                     compression_codec_name=compression_codec_name,
+                                                     max_dictionary_keys=max_dictionary_keys,
+                                                     for_read=False)
+
+    table_definition = None
+    if col_definitions is not None:
+        table_definition = _JTableDefinition.of([col.j_column_definition for col in col_definitions])
+
+    try:
+        if table_definition:
+            if write_instructions:
+                _JParquetTools.writeTable(table.j_table, destination, table_definition, write_instructions)
+            else:
+                _JParquetTools.writeTable(table.j_table, _get_file_object(destination), table_definition)
+        else:
+            if write_instructions:
+                _JParquetTools.writeTable(table.j_table, _get_file_object(destination), write_instructions)
+            else:
+                _JParquetTools.writeTable(table.j_table, destination)
+    except Exception as e:
+        raise DHError(e, "failed to write to parquet data.") from e
+
+
+def write_tables(tables: List[Table], destinations: List[str], col_definitions: List[Column],
+                 col_instructions: List[ColumnInstruction] = None, compression_codec_name: str = None,
+                 max_dictionary_keys: int = None, grouping_cols: List[str] = None):
+    """ Writes tables to disk in parquet format to a supplied set of destinations. If you specify grouping columns,
+    there must already be grouping information for those columns in the sources. This can be accomplished with
+    .groupBy(<grouping columns>).ungroup() or .sort(<grouping column>).
```

Comment:
These is just carrying over the javadoc for the java code as pydoc:

https://github.com/deephaven/deephaven-core/blob/1dc07f3c939e4a46c6ec698eb3cafcf0b0a4a254/extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java#L286

Any changes we make should be reflected there first (and sadly, copied over here since this is not auto-generated).

But either way, this PR doesn't look like the right place for that change.

Reply:
This PR is addressing the Python v2 API.  If the API or API docs are weird, those comments belong in this PR.  Once v2 goes out, we are more stuck with what we have.